app/grandchallenge/evaluation/models.py

```diff
@@ -1710,6 +1720,74 @@ class EvaluationGroundTruthGroupObjectPermission(GroupObjectPermissionBase):
     )
 
 
+class EvaluationManager(ComponentJobManager):
+    def get_evaluations_with_same_inputs(
+        self,
+        *,
+        inputs,
+        submission,
+        method,
+        ground_truth,
+        time_limit,
+        requires_gpu_type,
+        requires_memory_gb,
+    ):
+        existing_civs = self.retrieve_existing_civs(civ_data=inputs)
+        unique_kwargs = {
+            "submission": submission,
+            "method": method,
+            "time_limit": time_limit,
+            "requires_gpu_type": requires_gpu_type,
+            "requires_memory_gb": requires_memory_gb,
+        }
+
+        if ground_truth:
+            unique_kwargs["ground_truth"] = ground_truth
+        else:
+            unique_kwargs["ground_truth__isnull"] = True
+
+        input_interface_count = len(inputs)
+        configured_input_interface_slugs = submission.phase.inputs.values_list(
+            "slug", flat=True
+        )
+
+        # annotate the number of inputs and the number of inputs that match
+        # the existing civs and filter on both counts so as to not include evaluations
+        # with partially overlapping inputs
+        # or evaluations with more inputs than the existing civs
+        existing_evaluations = (
+            Evaluation.objects.filter(**unique_kwargs)
+            .annotate(
+                additional_input_count=Coalesce(
+                    Count(
+                        "inputs",
+                        filter=(
+                            Q(
+                                inputs__interface__slug__in=configured_input_interface_slugs
+                            )
+                        ),
+                        distinct=True,
+                    ),
+                    0,
+                ),
```

Comment:
I can't say that I fully understand this, but it appears that annotations with zero counts need special treatment for the filter below to work properly. I tried various things, but the `Coalesce` was ultimately the cleanest solution I could find. 

This also needs to be fixed in `annotate_input_output_counts` which we use for Jobs. It's not been an issue since we currently never pass empty inputs to this function and when we pass empty outputs, we don't filter on those subsequenty. But it should of course be fixed nonetheless.

Reply:
I think we've seen this before as empty matches will return `None`. Can't you use `Count(..., default=0)`? https://docs.djangoproject.com/en/5.1/ref/models/querysets/#default

Reply:
For count the `default` argument is not supported, see the note on the page you linked.

Reply:
Ah okay. Then the behaviour is strange, as they set `empty_result_set_value = 0`.

Reply:
Indeed, I don't quite understand what's going on either. I can see that the count annotation is 0, but then when I filter 0 != 0 for some reason. I thought it was a type difference, but the Count outputs an Integer, so that can't be it.

Reply:
Weird, surprised that we've not hit this before as we've used this pattern before, but maybe it's only been when the filter count is non-zero?

Reply:
Indeed, I think in all other places we either don't end up with zero-counts, or if we do (it can happen for `annotate_input_output_counts` when we don't pass in outputs), we at least don't filter on those.

Reply:
Good to know! Please add this to our knowledge base somewhere as this is a strange gotcha.